.github/workflows/golangci-feature.yml

```diff
@@ -0,0 +1,31 @@
+# golangci-feature runs on pull requests from branches that do not target main.
+#
+# Working on feature branches (PRs where the PR base-ref != main) is a common
+# workflow used, in order to make the merging of PRs easier certain lints are excluded
+# when it makes sense. Currently, unused lints are excluded since many PRs will add
+# unused code that will be used in a later PR.
+name: golangci-lint feature branch
+on:
+  pull_request:
+    # Ignore if the target is main. (Negation of golanci-lint.yml)
+    branches-ignore:
+      - main
```

Comment:
only run on pull requests not targeting main: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-excluding-branches